Work Submission

AnanthVivekanand opened the pull request.

AnanthVivekanand committed:
Added basic work submission

AnanthVivekanand committed:
Cleaned up files

AnanthVivekanand committed:
Improved work submission

AnanthVivekanand committed:
Removed unneeded index.js

AnanthVivekanand committed:
Updated README.md for Work Submission

Changed example to include options such as onSubmitWorkSuccess and onSubmitWorkFail

arnabk commented on src/index.js:
```diff
@@ -12,9 +13,14 @@ const defaultConfig = {
 class Client {
-
+submit() {
+var args = Array.prototype.slice.call(arguments);
```
I think we just do `submit(...args)` instead of using slice.
I am not sure why the code formatting is messed up all over. Could you please check?

AnanthVivekanand commented:
Ah, I was trying to prepend `client` (the socket) to the arguments. That way, when the arguments are forwarded to submitWork.js, it will have access to the socket in order to send data to the pool. Without adding `client` to the arguments, it won't be able to access the socket.

It works fine, do you prefer a different method? 

```
submit() {
    var args = Array.prototype.slice.call(arguments);
    args.unshift(this.client); //Adds client to the arguments
    submitWork.apply(this, args); //Forwards the arguments (and client) to submitWork.js
}
```

AnanthVivekanand committed:
Fixed formatting

arnabk commented:
@KorkyMonster There are couple of things to consider.

- It looks like you are using an IDE which uses 4 spaces as tab. This project has `.editorconfig` which defines what this project is formatted with. It would be better if your IDE uses that instead of overriding it.
- I took the time to look into your code further. I think it would be better if you define what arguments you are passing instead of `...args`. Using `arguments` is powerful but not very useful to keep the code readability. Also, instead of passing each argument as separate parameter, pass them in one object like opts. That way, the order of the argument wont matter. Generally what I do is, when the list of parameter is 2 or less, I pass them as separate parameter, but when it grows, its better to use `opt` for flexibility later.

AnanthVivekanand committed:
Fixed indention from 4 to 2 spaces

AnanthVivekanand committed:
Changed work submission parameters

arnabk commented on src/processData.js:
```diff
@@ -2,75 +2,86 @@ const extend = require('lodash/extend');
+    onSubmitWorkSuccess,
+    onSubmitWorkFail,
+  } = updatedOptions;
+  console.log(jsonData);
```
Please remove this `console.log` statement

arnabk commented on src/submitWork.js:
```diff
@@ -0,0 +1,7 @@
+  submitWork
+} = require('./messageContants');
+
+module.exports = (options, client) => {
```
Once you do the above changes, client will be part of options.

arnabk commented on src/index.js:
```diff
@@ -12,9 +13,12 @@ const defaultConfig = {
 
 class Client {
-
+  submit(options) {
```
Can we  change this code to use 
```
submit(options) {
  submitWork({
     ...options,
     client, 
  });
}
```

arnabk commented:
If you are interested, you could use destructing in lot of places and improve the code. If not, I will do it in separate PR.

AnanthVivekanand committed:
Destructured work submission

arnabk commented on src/index.js:
```diff
@@ -14,7 +14,10 @@ const defaultConfig = {
-    submitWork.apply(this, [options, client]);
+    submitWork([
+      ...options,
+      this.client,
```
This won't work. You should add following code in the top of the module

```
const client = new net.Socket();
client.setEncoding('utf8');
```
instead of `start` method

arnabk commented on src/submitWork.js:
```diff
@@ -2,6 +2,6 @@ const {
-module.exports = (options, client) => {
-  client.write(submitWork.replace("<worker.name>", options.worker_name).replace("<jobID>", options.job_id).replace("<ExtraNonce2>", options.extranonce2).replace("<ntime>", options.ntime).replace("<nonce>", options.nonce));
+module.exports = (options) => {
+  options[5].write(submitWork.replace("<worker.name>", options[0]).replace("<jobID>", options[1]).replace("<ExtraNonce2>", options[2]).replace("<ntime>", options[3]).replace("<nonce>", options[4]));
```
This is bad. Use object destructing.

AnanthVivekanand committed:
Ok, I think I finally figured it out

AnanthVivekanand commented:
Is this project still being maintained? PR has been made a while ago to solve issue #2.

arnabk commented on src/connect.js:
```diff
@@ -1,10 +1,16 @@
-const { subscribe } = require('./messageContants');
+const {
```
This is an unnecessary change. Please revert changes to this file

AnanthVivekanand replied:
Changes have been reverted.

arnabk commented on src/onError.js:
```diff
@@ -1,12 +1,15 @@
 module.exports = (client, options, error) => {
-  const { autoReconnectOnError, onError } = options;
+  const {
+    autoReconnectOnError,
```
Please revert changes to this file.

AnanthVivekanand replied:
Changes have been reverted.

arnabk commented:
It is. You have to follow the coding standards of the project and do not submit unnecessary formatting. Please go over list of comments and do the changes accordingly.

AnanthVivekanand committed:
Cleaned up unnecessary formatting

AnanthVivekanand commented:
I've cleaned up several cases of unnecessary formatting. Thanks for letting me know about that.

I've left `src/processData.js` with some formatting changes because `.editorconfig` declares this:
```
indent_style = space
indent_size = 2
```

I've changed the instances of 4 spaces per indent in that files to 2, making it adhere to .editorconfig. Please look over the changes I've made and let me know of what you think.

AnanthVivekanand commented on src/processData.js:
```diff
@@ -2,75 +2,86 @@ const extend = require('lodash/extend');
-            client.write(authorize.replace("<worker.name>", worker).replace("<worker.pass>", defaultTo(password, 'x')));
-        }
-        break;
-      case miningNotify:
```
I've changed instances of 4 spaces per indent, like that below, to two space per indent.

arnabk commented:
@KorkyMonster Looks good to me. Have you tested connecting to a pool and making sure that it is receiving work and is also able to submit work? If yes, then please run `npm version minor` and push the changes. I will merge and release it.

Thank you

AnanthVivekanand committed:
1.1.0

AnanthVivekanand commented:
Published! 

Here's a javascript miner written with `stratum-client`: https://github.com/KorkyMonster/garlicoin-miner-nodejs/

arnabk commented:
Fixes https://github.com/arnabk/stratum-client/issues/2
